src/datomish/db.cljc

```diff
@@ -67,4 +635,391 @@
   [db find args]
   (a/reduce (partial reduce-error-pair conj) [[] nil]
             (<?run db find args)))
-    
+
+
+(defn <resolve-lookup-refs [db report]
+  {:pre [(db? db) (report? report)]}
+
+  (go-pair
+    (->>
+      (vec (for [[op & entity] (:entities report)]
+             (into [op] (for [field entity]
+                          (if (lookup-ref? field)
+                            (first (<? (<eavt db field))) ;; TODO improve this -- this should be avet, shouldn't it?
```

Comment:
Yeah -- I should really just remove this, since it's not complete and not tested.  I might get to it before I go.